modules/fleet/src/javaRestTest/java/org/elasticsearch/fleet/FleetSystemIndexIT.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.fleet;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.test.rest.ESRestTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class FleetSystemIndexIT extends ESRestTestCase {
+
+    private final String indexName;
+
+    public FleetSystemIndexIT(@Name("indexName") String indexName) {
+        this.indexName = indexName;
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(
+            new Object[] { ".fleet-outputs" },
+            new Object[] { ".fleet-policies" },
+            new Object[] { ".fleet-policies-enrollment-keys" },
+            new Object[] { ".fleet-policies-inputs" },
+            new Object[] { ".fleet-agents" },
+            new Object[] { ".fleet-agents-checkins" }
```

Comment:
These have changed slightly the following indexes are needed:

```
.fleet-servers
.fleet-policies
.fleet-policies-leader
.fleet-agents
.fleet-actions
.fleet-actions-results
```

As for the Agent check-ins that will occur in `logs-elastic_agent.checkin-default`. This will follow the indexing strategy of Fleet and will be a datastream. I don't know if this needs to be registered in this plugin or not?

Reply:
Thanks, I'll update the indices shortly.

The dividing line is "Should a user ever directly query, visualize, and/or write to this index?" - if the answer is no, then it should probably be a system index. There's also a practical consideration in that there's no such thing as a "system data stream" (currently), so if being a data stream is a requirement, it can't be a system index.

Reply:
I've updated the plugin to use this list and removed the ones which were not on the list.

Reply:
A user should be able to directly query and visualize the `logs-elastic_agent.checkin-default`.